hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SynchronousQueueStreamSerializer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import java.util.Collection;
+import java.util.concurrent.SynchronousQueue;
+
+/**
+ * The {@link SynchronousQueue} serializer
+ */
+public class SynchronousQueueStreamSerializer extends AbstractCollectionStreamSerializer {
+    @Override
+    protected Collection createCollection(int size) {
+        return new SynchronousQueue();
```

Comment:
Also, we can leave a comment in javadoc/ref manual that we don't copy the `fair` boolean parameter.